tests/plugins/widget/_helpers/tools.js

```diff
@@ -32,7 +33,10 @@ var widgetTestsTools = ( function() {
 					loaded: function( evt ) {
 						editor = evt.editor;
 
-						initialData = fixHtml( editor.getData(), config.ignoreStyle );
+						var ignoreStyle = config.ignoreStyle;
+						ignoreStyle = typeof ignoreStyle === 'function' ? ignoreStyle( this ) : ignoreStyle;
+
+						initialData = fixHtml( editor.getData(), ignoreStyle );
```

Comment:
You implemented this feature very high in function aggregation. `ignoreStyle` option for `addTests` function is not used very extensively, AFAIK only for `mathjax-mock` tests. However, `fixHtml` function is used in many places where this improvement may be very useful. Why not move the whole logic into `fixHtml` function itself?

Of course, it makes sense to still pass `editor` instance when possible (for `addTests` etc) to avoid some creepy workarounds when writing tests. Unfortunately, our `CKEDITOR.tools.bind` the function doesn't work correctly, as it doesn't allow to bind function arguments and would make implementation much cleaner

 We have already nice PR for this feature https://github.com/ckeditor/ckeditor-dev/pull/2862 and maybe it's a good time to finish it? I will bump this PR a bit.